dapp/pages/api/stats.js

```diff
@@ -0,0 +1,58 @@
+import { ethers } from 'ethers'
```

Comment:
Having a stats page hosted on the DApp works.

Though I wonder if it would make more sense to move this API to the [ousd analytics stack](https://github.com/OriginProtocol/ousd-analytics) since that's where we are putting all the stats related APIs.
Also an advantage of doing that is that we would be able to differentiate the user traffic hitting the dapp vs coming from the website on our Alchemy dashboard because we use different Alchemy keys for dapp vs analytics.

CC @mikeshultz @DanielVF for their opinion

Reply:
We could definitely do this on the analytics side and keep the contracts simpler. That's almost always my vote on that trade off. Also we don't have to spend extra gas to increment the counter.

Reply:
This here is just 2 client-side JSON-RPC calls so I don't really think this warrants a lot of debate.  This isn't currently available via analytics so maybe just roll with this for now.

I am planning to add staking data to ousd-analytics soon, though.  We could add API endpoints for this data if you all find it useful.

Reply:
Ah, I missed the contract changes necessary for this.  I don't feel super strongly either way then.  I can prioritize work on the analytics collection to get it ready for release, but maybe having that counter in-contract isn't a bad idea either.

Reply:
Yeah, it would anyway be useful to have the staking data in the analytics DB so that the product/business team can analyze it.

If we would capture in the DB the [Staked and Withdrawn events](https://github.com/OriginProtocol/origin-dollar/blob/master/contracts/contracts/staking/SingleAssetStaking.sol#L458) emitted by the contract I think that this by itself should be enough to calculate the number of stakers.

If we want to go beyond, upon seeing these events, we could call the contract's getAllStakes() method and store/update in the DB the details of all the stakes for the account that triggered the event.